go-controller/pkg/ovn/gateway_init.go

```diff
@@ -258,6 +281,80 @@ func gatewayInit(nodeName string, clusterIPSubnet []*net.IPNet, hostSubnets []*n
 			"stderr: %q, error: %v", gatewayRouter, stdout, stderr, err)
 	}
 
+	if hybridGatewayConfig != nil {
+		hybridExternalSwitch := hybridSwitchPrefix + nodeName
+		// hybrid GR connection
+		cmdArgs = []string{
+			"--", "--if-exists", "lrp-del", gwRouterToHybridSwitchPrefix + gatewayRouter,
+			"--", "lrp-add", gatewayRouter, gwRouterToHybridSwitchPrefix + gatewayRouter,
+			hybridGatewayConfig.MACAddress.String(),
+		}
+		for _, ip := range hybridGatewayConfig.IPAddresses {
+			cmdArgs = append(cmdArgs, ip.String())
+		}
+		cmdArgs = append(cmdArgs,
+			"--", "set", "logical_router_port", gwRouterToHybridSwitchPrefix+gatewayRouter,
+			"external-ids:gateway-physical-ip=yes")
+
+		stdout, stderr, err = util.RunOVNNbctl(cmdArgs...)
+		if err != nil {
+			return fmt.Errorf("failed to add logical port to router %s, stdout: %q, "+
+				"stderr: %q, error: %v", gatewayRouter, stdout, stderr, err)
+		}
+
+		// Connect the external_switch to the router.
+		stdout, stderr, err = util.RunOVNNbctl("--", "--may-exist", "lsp-add",
+			hybridExternalSwitch, hybridSwitchToGwRouterPrefix+gatewayRouter, "--", "set",
+			"logical_switch_port", hybridSwitchToGwRouterPrefix+gatewayRouter, "type=router",
+			"options:router-port="+gwRouterToHybridSwitchPrefix+gatewayRouter,
+			"addresses="+"\""+hybridGatewayConfig.MACAddress.String()+"\"")
+		if err != nil {
+			return fmt.Errorf("failed to add logical port to router %s, stdout: %q, "+
+				"stderr: %q, error: %v", gatewayRouter, stdout, stderr, err)
+		}
+
+		// add policy route to force host network destined traffic through br-local
+		for _, addr := range hybridGatewayConfig.IPAddresses {
+			var l3Prefix, localNextHop string
+			if utilnet.IsIPv6(addr.IP) {
+				l3Prefix = "ip6"
+				// find br-local next hop
+				for _, nextHop := range l3GatewayConfig.NextHops {
+					if utilnet.IsIPv6(nextHop) {
+						localNextHop = nextHop.String()
+						break
+					}
+				}
+			} else {
+				l3Prefix = "ip4"
+				for _, nextHop := range l3GatewayConfig.NextHops {
+					if nextHop.To4() != nil {
+						localNextHop = nextHop.String()
+						break
+					}
+				}
+			}
+			if localNextHop == "" {
+				return fmt.Errorf("unable to find br-local next hop for ip type: %s, ips: %v", l3Prefix,
+					l3GatewayConfig.NextHops)
+			}
+			// IPNet passed into annotation is currently wrong. It is a full IP/mask like 169.55.22.2/24 instead of
+			// 169.55.22.0/24 so need to get proper ipMask
```

Comment:
It's not _wrong_, it's just not the piece of information you want. There are places where we need it in exactly the form it is now

Reply:
OK I'll fix the comment. When I wrote this I thought that the annotation should be the network cidr and we were accidentally setting it as an ip addr + mask.